src/cpp/ext/csm/metadata_exchange.cc

```diff
@@ -364,13 +391,13 @@ ServiceMeshLabelsInjector::ServiceMeshLabelsInjector(
       opentelemetry::sdk::resource::SemanticConventions::kK8sNamespaceName);
   absl::string_view cluster_name_value = GetStringValueFromAttributeMap(
       map, opentelemetry::sdk::resource::SemanticConventions::kK8sClusterName);
-  absl::string_view cluster_location_value = GetStringValueFromAttributeMap(
+  absl::string_view location_value = GetStringValueFromAttributeMap(
       map, opentelemetry::sdk::resource::SemanticConventions::
-               kCloudRegion);  // if regional
-  if (cluster_location_value == "unknown") {
-    cluster_location_value = GetStringValueFromAttributeMap(
+               kCloudAvailabilityZone);  // if zonal
```

Comment:
Looks like zone is taking precedence over region now.  Is that switch intentional?

Reply:
Yeah, if everything works normally, only one of `zone`/`region` would be set, and the precedence wouldn't matter, but my reading from internal threads said that if for some reason both `zone` and `region` are set, then `zone` is more specific than `region`.